### PR TITLE
Fix PE-D Bugs

### LINKED ISSUE #240: [PE-D][Tester A] Volunteer Free invalid command format error message does not include all parameters
Error message is missing the index parameter

-------------
Labels: `type.FeatureFlaw` `severity.VeryLow`
original: FriedCabbageSalad/ped#10

### LINKED ISSUE #241: [PE-D][Tester B] No mention of duplicate names not being allowed for both volunteers and events
Not mentioned in the user guide, found out during testing. 

-------------
Labels: `severity.Medium` `type.DocumentationBug`
original: VedJoshi/ped#13

### LINKED ISSUE #242: [PE-D][Tester A] Volunteer Unfree invalid command format error message does not include all parameters
Error message is missing the index parameter

-------------
Labels: `type.FeatureFlaw` `severity.VeryLow`
original: FriedCabbageSalad/ped#11

### LINKED ISSUE #248: [PE-D][Tester B] Missing Use Cases
Extension of duplicate name not mentioned.

-------------
Labels: `severity.Low` `type.DocumentationBug`
original: VedJoshi/ped#17

### LINKED ISSUE #249: [PE-D][Tester B] Unrendered UML
UM Use Case Diagram is not rendered correctly in the PDF.

-------------
Labels: `severity.High` `type.DocumentationBug`
original: VedJoshi/ped#18

### LINKED ISSUE #257: [PE-D][Tester D] [DG] Broken UML Diagram
UML Use Case Diagram (After all the Use Cases) is broken.

-------------
Labels: `severity.High` `type.DocumentationBug`
original: elliot-tang/ped#7

### LINKED ISSUE #259: [PE-D][Tester D] [DG: Use Cases] Missing Use Cases for Several Features
Missing use cases for several commands, including Free/Unfree, Filter, Delete etc, each of which have extensions/more detailed MSS which should be covered.

-------------
Labels: `severity.Low` `type.DocumentationBug`
original: elliot-tang/ped#9

### LINKED ISSUE #269: [PE-D][Tester A] Non-Functional Requirements having extra audience that are not mentioned in UG or the App
Donors are not involved anywhere in the app or UG, yet are mentioned to have an impact on app performance

-------------
Labels: `type.DocumentationBug` `severity.VeryLow`
original: FriedCabbageSalad/ped#23

### PR DESCRIPTION
**Pull request by Fluffykan**
Fixes #269 
Fixes #259
Fixes #257 
Fixes #249 
Fixes #248 
Fixes #242
Fixes #241
Fixes #240

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -220,62 +220,136 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
-   - 4ai. System notifies user and the user can edit the event details, returning to step 2 afterward.
-     Use Case Ends.
+- 1a. Volunteer does not exist in the system.
+    - 1ai. System notifies user, and prompts the user to select a valid volunteer. Returns to step 1.
 
-**Guarantees**:
-- New volunteer is stored in the system if all required information is present and valid.
+#### UC04. Add a Free Day to a Volunteer
+
+**Description**: Add a free day to a volunteer's schedule.
```
[the remark this answers is not in the document]
Comment (Fluffykan) [reply docs/DeveloperGuide.md:287]: You're right, I've amended it

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -220,62 +220,136 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+**Preconditions**: Volunteer must exist in the system.
+
+**MSS**:
+
+1. User selects the volunteer to remove a free day from.
+2. User selects the date(s) to remove as a free day.
+3. User submits the information to the system.
+4. System checks if the volunteer is available on the selected date(s).
+5. System removes the free day from the volunteer's schedule and confirms removal.<br/>
+   Use Case Ends.
```
[the remark this answers is not in the document]
Comment (Fluffykan) [reply docs/DeveloperGuide.md:329]: Amended!

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -220,62 +220,136 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+      Use Case Ends.<br><br>
+
+#### UC06. Assign Volunteer to Event
 
 **Description**: Assign a volunteer to a specific event.
 
-**Preconditions**:
-- Event exists in the system.
-- Volunteer is registered in the system.
+**Preconditions**: Event and Volunteer exists in the system.
```
[the remark this answers is not in the document]
Comment (Fluffykan) [reply docs/DeveloperGuide.md:352]: Amended!

**Review by JovenSoh — APPROVED** [review]
LGTM

**Pull request merged**